src/hooks/useLeaderboard.tsx

```diff
@@ -4,23 +4,46 @@ import { useToast } from '@chakra-ui/react';
 import { getLeaderboardDataHandlerFromType } from '../services/leaderboard';
 import { assertTypeAndOrderingIntegrity } from '../utils';
 
+interface CacheEntry {
+  data: LeaderboardEntry[];
+  timestamp: number;
+}
+
+const leaderboardCache = new Map<string, CacheEntry>();
```

Comment:
maybe we should persist across refresh. did you happen to check before and after how many network calls you were making? i assume this just prevents rerequesting when switching back and forth between order/type, but not page refresh